src/backend/TrafficCourts/Staff.Service/Services/ComsService.cs

```diff
@@ -75,22 +86,12 @@ public async Task DeleteFileAsync(Guid fileId, CancellationToken cancellationTok
     {
         _logger.LogDebug("Searching files through COMS");
 
-        Dictionary<Guid, string> fileData = new();
-
         FileSearchParameters searchParameters = new(null, metadata, tags);
 
-        List<FileSearchResult> searchResult = await _objectManagementService.FileSearchAsync(searchParameters, cancellationToken);
+        IList<FileSearchResult> searchResult = await _objectManagementService.FileSearchAsync(searchParameters, cancellationToken);
 
-        foreach (var result in searchResult)
-        {
-            result.Metadata.TryGetValue("name", out string? filename);
-            if (string.IsNullOrEmpty(filename))
-            {
-                filename = "unknown";
-                _logger.LogDebug("name value from metadata is empty");
-            }
-            fileData.Add(result.Id, filename);
-        }
+        Dictionary<Guid, string> fileData = searchResult
+            .ToDictionary(file => file.Id, file => file.FileName ?? "unknown");
```

Comment:
Should we also check if the `searchResult` is empty here before mapping it to the dictionary to return?

Reply:
The resulting dictionary would be empty if `searchResult` does not have any items.